packages/core/core-flows/src/order/steps/cancel-order-change.ts

```diff
@@ -0,0 +1,38 @@
+import { CancelOrderChangeDTO, IOrderModuleService } from "@medusajs/types"
+import { ModuleRegistrationName } from "@medusajs/utils"
+import { createStep, StepResponse } from "@medusajs/workflows-sdk"
+
+export const cancelOrderChangeStepId = "cancel-order-change"
+export const cancelOrderChangeStep = createStep(
+  cancelOrderChangeStepId,
+  async (data: CancelOrderChangeDTO, { container }) => {
+    const service = container.resolve<IOrderModuleService>(
+      ModuleRegistrationName.ORDER
+    )
+
+    const dataBeforeUpdate = await service.retrieveOrderChange(data.id, {
+      select: ["status", "metadata"],
+    })
+
+    await service.cancelOrderChange(data)
+
+    return new StepResponse(void 0, {
+      id: data.id,
+      status: dataBeforeUpdate.status,
+      canceled_at: null,
+      canceled_by: null,
+      metadata: dataBeforeUpdate.metadata,
```

Comment:
do we need the metadata here?
that probably won't be touched if we only update the other fields no?

Reply:
good catch, made it dynamic based on inputs